src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderSentence.kt

```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa
+
+import jakarta.persistence.Column
+import jakarta.persistence.Embeddable
+import jakarta.persistence.EmbeddedId
+import jakarta.persistence.Entity
+import jakarta.persistence.FetchType
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.Table
+import org.hibernate.Hibernate
+import java.io.Serializable
+
+@Embeddable
+data class SentenceId(
+  @ManyToOne(optional = false, fetch = FetchType.LAZY)
+  @JoinColumn(name = "OFFENDER_BOOK_ID", nullable = false)
```

Comment:
Do you need nullable = true for when we move to Hibernate 6? (remember the issue Peter found) https://github.com/ministryofjustice/hmpps-nomis-prisoner-api/pull/145#pullrequestreview-1225939691

Reply:
I just don't know - I suspect not since the problem we had was for a foreign key where actually other entity didn't exist, in other words there was no foreign key constraint and we had bad data